### ISSUE TITLE
yolox backbone error

### ISSUE THREAD

**Issue by ZhuXiongwei24** [zgcr/SimpleAICV-pytorch-ImageNet-COCO-training#25]
The yolox backbone in this codebase without focus operation, the shape of stem between https://github.com/Megvii-BaseDetection/YOLOX and this codebase is different.
The stem of yolox_m backbone in https://github.com/Megvii-BaseDetection/YOLOX:
![image](https://user-images.githubusercontent.com/22865947/179176137-f2d05e66-e158-469c-ad00-79fcfe204ac0.png)
The stem of yolox_m backbone in this codebase:
![image](https://user-images.githubusercontent.com/22865947/179176247-09bed639-fdb1-44c5-ae28-20887ce25abd.png)

class Focus(nn.Module):
    """Focus width and height information into channel space."""

    def __init__(self, in_channels, out_channels, ksize=1, stride=1, act="silu"):
        super().__init__()
        self.conv = BaseConv(in_channels * 4, out_channels, ksize, stride, act=act)

    def forward(self, x):
        # shape of x (b,c,w,h) -> y(b,4c,w/2,h/2)
        patch_top_left = x[..., ::2, ::2]
        patch_top_right = x[..., ::2, 1::2]
        patch_bot_left = x[..., 1::2, ::2]
        patch_bot_right = x[..., 1::2, 1::2]
        x = torch.cat(
            (
                patch_top_left,
                patch_bot_left,
                patch_top_right,
                patch_bot_right,
            ),
            dim=1,
        )
        return self.conv(x)

**Comment by zgcr** [comment]
s.



> The yolox backbone in this codebase without focus operation, the shape of stem between https://github.com/Megvii-BaseDetection/YOLOX and this codebase is different. The stem of yolox_m backbone in https://github.com/Megvii-BaseDetection/YOLOX: ![image](https://user-images.githubusercontent.com/22865947/179176137-f2d05e66-e158-469c-ad00-79fcfe204ac0.png) The stem of yolox_m backbone in this codebase: ![image](https://user-images.githubusercontent.com/22865947/179176247-09bed639-fdb1-44c5-ae28-20887ce25abd.png)
> 
> class Focus(nn.Module): """Focus width and height information into channel space."""
> 
> ```
> def __init__(self, in_channels, out_channels, ksize=1, stride=1, act="silu"):
>     super().__init__()
>     self.conv = BaseConv(in_channels * 4, out_channels, ksize, stride, act=act)
> 
> def forward(self, x):
>     # shape of x (b,c,w,h) -> y(b,4c,w/2,h/2)
>     patch_top_left = x[..., ::2, ::2]
>     patch_top_right = x[..., ::2, 1::2]
>     patch_bot_left = x[..., 1::2, ::2]
>     patch_bot_right = x[..., 1::2, 1::2]
>     x = torch.cat(
>         (
>             patch_top_left,
>             patch_bot_left,
>             patch_top_right,
>             patch_bot_right,
>         ),
>         dim=1,
>     )
>     return self.conv(x)
> ```

Hi,due to the lack of GPU resources，I didn't fully reproduce yolox.I'm not sure yolov3/yolov5/yolox codes are completely correct.I recommend that you use algorithms for which I have provided training results.
